mkdocs_monorepo_plugin/plugin.py

```diff
@@ -64,25 +68,28 @@ def on_pre_page(self, page, config, files):
         return page
 
     def on_serve(self, server, config, **kwargs):
-        # Support mkdocs < 1.2
-        if hasattr(server, 'watcher'):
-            buildfunc = list(server.watcher._tasks.values())[0]['func']
-
-            # still watch the original docs/ directory
-            server.watch(self.originalDocsDir, buildfunc)
-
-            # watch all the sub docs/ folders
-            for _, docs_dir, yaml_file in self.resolvedPaths:
-                server.watch(yaml_file, buildfunc)
-                server.watch(docs_dir, buildfunc)
-        else:
-            # still watch the original docs/ directory
-            server.watch(self.originalDocsDir)
-
-            # watch all the sub docs/ folders
-            for _, docs_dir, yaml_file in self.resolvedPaths:
-                server.watch(yaml_file)
-                server.watch(docs_dir)
+        # Watch extra files only if this plugin was actually initialized with
+        # `on_config`
+        if self.originalDocsDir is not None:
```

Comment:
nit: couldn't you reduce diff here (and follow the pattern in `.on_config` if you just start with:

```py
if self.originalDocsDir is None:
  return
```

Reply:
that's so true, and a great take on it. the only qualm i had about it was that i was worried about being "pythonic" or whatever. that being said i'll probably take your solution